core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -3482,6 +3482,18 @@ private RecordCursorFactory generateOrderBy(
                     final CharSequence column = orderByColumnNames.getQuick(i);
                     int index = metadata.getColumnIndexQuiet(column);
 
+                    if (index < 0) {
```

Comment:
this looks like a collateral change. Generator should not experience this situation. The "optimiser" usually performs validation is user feedback. Dealing with the error here could be hiding deficiencies elsewhere